dask/base.py

```diff
@@ -0,0 +1,85 @@
+import warnings
+from operator import attrgetter
+
+from toolz import merge, groupby, unique
+
+from .context import _globals
+
+
+class Base(object):
+    """Base class for dask collections"""
+
+    def visualize(self, optimize_graph=False):
+        from dask.dot import dot_graph
+        if optimize_graph:
+            return dot_graph(self._config.optimize(self.dask, self._keys()))
+        else:
+            return dot_graph(self.dask)
+
+    def _visualize(self, optimize_graph=False):
+        warn = DeprecationWarning("``_visualize`` is deprecated, use "
+                                  "``visualize`` instead.")
+        warnings.warn(warn)
+        return self.visualize(optimize_graph)
+
+    def compute(self, **kwargs):
+        return compute(self, **kwargs)[0]
+
+
+class Config(object):
+    """Specifies a configuration for a dask collection.
+
+    This holds functions that are used to specialize dask schedulers for a
+    specific collection.
+
+    Parameters
+    ----------
+    optimize : callable
+        Takes a dask and iterable of keys, and returns an optimized dask.
+    default_get : callable
+        The default scheduler to use (e.g. ``dask.threaded.get``)
+    finalize : callable
+        Takes a dask collection and an iterable of results, and converts them
+        into the desired python type.
+    """
+    def __init__(self, optimize, default_get, finalize):
+        self.optimize = optimize
+        self.default_get = default_get
+        self.finalize = finalize
```

Comment:
Aesthetically I really like that I can describe `Array` as just a class with a dask graph and a compute method.  Adding `Base` hurts a little bit but it seems right.  Adding another encapuslated `Config` object hurts more.  It may be the right thing to do but I plan to be annoyingly picky about it for a while.